src/sentry/notifications/utils/__init__.py

```diff
@@ -132,30 +136,71 @@ def get_rules(
     ]
 
 
+def process_serialized_committers(
+    committers: Sequence[AuthorCommitsSerialized], commits: MutableMapping[str, Mapping[str, Any]]
+) -> MutableMapping[str, Mapping[str, Any]]:
+    """
+    Transform committer data from nested structure to flat commit dictionary.
+
+    Args:
+        committers: List of {author, commits}
+        commits: Dict to store processed commits (modified in-place)
+
+    Returns:
+        Dict mapping commit IDs to enriched commit data with shortId, author, subject
+    """
+    for committer in committers:
+        for commit in committer["commits"]:
+            if commit["id"] not in commits:
+                commit_data = dict(commit)
+                commit_data["shortId"] = commit_data["id"][:7]
+                commit_data["author"] = committer["author"]
+                commit_data["subject"] = (
+                    commit_data["message"].split("\n", 1)[0] if commit_data["message"] else ""
+                )
+                if commit.get("pullRequest"):
+                    commit_data["pull_request"] = commit["pullRequest"]
+                commits[commit["id"]] = commit_data
+    return commits
+
+
+def get_suspect_commits_by_group_id(
+    project: Project,
+    group_id: int,
+) -> Sequence[Mapping[str, Any]]:
+    """
+    Get suspect commits for workflow notifications that only have a group ID (no Event).
+
+    Uses Commit Context (SCM integrations) only - no release-based fallback.
+    Returns commits sorted by recency, not suspicion score.
+    """
+    commits: MutableMapping[str, Mapping[str, Any]] = {}
+    try:
+        committers = get_serialized_committers(project, group_id)
+    except (Commit.DoesNotExist, Release.DoesNotExist):
+        pass
+    except Exception as exc:
+        logging.exception(str(exc))
```

Comment:
you can just do `sentry_sdk.capture_exception(exc)` instead of logging the stringified exception